Mathlib/Algebra/CharP/ExpChar.lean

```diff
@@ -46,6 +46,32 @@ class inductive ExpChar (R : Type u) [Semiring R] : ℕ → Prop
 #align exp_char ExpChar
 #align exp_char.prime ExpChar.prime
 
+/-- The exponential characteristic is unique. -/
+theorem ExpChar.eq {p q : ℕ} (hp : ExpChar R p) (hq : ExpChar R q) : p = q := by
+  cases' hp with hp _ hp' hp
+  · cases' hq with hq _ hq' hq
+    exacts [rfl, False.elim (Nat.not_prime_zero (CharP.eq R hq (CharP.ofCharZero R) ▸ hq'))]
+  · cases' hq with hq _ hq' hq
+    exacts [False.elim (Nat.not_prime_zero (CharP.eq R hp (CharP.ofCharZero R) ▸ hp')),
+      CharP.eq R hp hq]
+
+variable {R} in
```

Comment:
On the other hand `R` should be explicit here, since `hq` is provided as an instance.